server.js

```diff
@@ -0,0 +1,219 @@
+import {PassThrough} from 'node:stream'
+import {URL, pathToFileURL} from 'node:url'
+
+import {unified} from 'unified'
+import {engine} from 'unified-engine'
+import {VFile} from 'vfile'
+import {
+  Diagnostic,
+  DiagnosticSeverity,
+  Position,
+  Range,
+  TextDocumentSyncKind,
+  TextEdit
+} from 'vscode-languageserver/node.js'
+
+/**
+ * Convert a unist point to a language server protocol position.
+ *
+ * @param {import('unist').Point} point
```

Comment:
Maybe move imports to the start?

Reply:
I’ll process do this for consistency, because I know this is how this is handled by most unified projects, but IMHO this results in lower quality code, especially in lower quality output.

`typedef` annotations are used to define and export a type. I think most types of the form I have seen so far actually `@typedef {import('package').MyType} MyType` shouldn’t be exported. Also unused typedefs are not reported (which is correct, as they are exported).

In this particular case we’re dealing with 2 text positioning type formats: `unist` and `vscode-languageserver-types`. I like how `@param {import('unist').Point}` namespaces this by default.

Reply:
- But this style is verbose, e.g., ` * @param {import('vscode-languageserver').TextDocuments<import('vscode-languageserver-textdocument').TextDocument>} documents`
- Tooling could help with checking unused imports

What is an isn’t exported is getting quite hard anyway, regardless of types: every internal module was accessible from the file-system before already, and now there are import/export maps and import/export conditions.
Having a single small `index.js` file that just re-exports the values and types that are considered the public API, might be a good idea even if `@typedef` worked differently?

Reply:
> * But this style is verbose, e.g., ` * @param {import('vscode-languageserver').TextDocuments<import('vscode-languageserver-textdocument').TextDocument>} documents`

Yes, this example is verbose. It’s also the only complex example for this project.

> * Tooling could help with checking unused imports

It can’t, because it’s exported, so technically not unused. `import()` defined an import, `@typedef` defines an export.

> What is an isn’t exported is getting quite hard anyway, regardless of types: every internal module was accessible from the file-system before already, and now there are import/export maps and import/export conditions. Having a single small `index.js` file that just re-exports the values and types that are considered the public API, might be a good idea even if `@typedef` worked differently?

I don’t think it’s that hard.

The following is not exported:

```js
function unistPointToLspPosition(point) {
  return Position.create(point.line - 1, point.column - 1)
}
```

The following is exported:

```js
export function unistPointToLspPosition(point) {
  return Position.create(point.line - 1, point.column - 1)
}
```

For types it’s the same.

The following contains no exports:

```js
/**
 * @param {import('unist').Point} point
 * @returns {Position}
 */
```

The following creates and exports `Point` type, which is an alias of `import('unist').Point`, but with JSDoc stripped. This means the packages now needs to depend on `@types/unist`.

```js
/**
 * @typedef {import('unist').Point} Point
 */

/**
 * @param {Point} point
 * @returns {Position}
 */
```